roles/ec2_networking_resources/tasks/create.yml

```
@@ -0,0 +1,80 @@
+---
+- name: Validate options
+  ansible.builtin.fail:
+    msg: "When creating resources, all of the following options must be provided: ec2_networking_resources_vpc_cidr_block, ec2_networking_resources_subnet_cidr_block, ec2_networking_resources_sg_internal_name"
+  when: ec2_networking_resources_vpc_cidr_block | default("", true) == "" or
+    ec2_networking_resources_subnet_cidr_block | default("", true) == "" or
+    ec2_networking_resources_sg_internal_name | default("", true) == ""
```

```suggestion
  when: ec2_networking_resources_vpc_cidr_block is undefined or ec2_networking_resources_subnet_cidr_block is undefined or ec2_networking_resources_sg_internal_name is undefined
```
what about using ``undefined`` to test the variable?

---

As is it also checks for empty values.